### PR TITLE
Support timeout for repo testing.

### LINKED ISSUE #47: staticanalysis smoke test harness should have hard timeout
Right now there's a few test repos we're using that hang indefinitely.  We should have a timeout before we automatically fail and move on, since the primary purpose here is breadth of repos.

### PR DESCRIPTION
**Pull request by danielmoy-google**
fixes #47.

### CODE REVIEW

File: kythe/go/extractors/config/smoke/smoke_test.go [hunk 1]
```diff
@@ -159,7 +160,7 @@ func verifyRepoCoverage(t *testing.T, te testExtractor, expectedDownloadCount, e
 		repoFetcher: te,
 	}
 
-	r, err := h.TestRepo("foo")
+	r, err := h.TestRepo(context.TODO(), "foo")
```
Comment (creachadair) [inline kythe/go/extractors/config/smoke/smoke_test.go]: Can you not pass along the caller's `ctx` here? (If not, please add a comment to say why)

Comment (danielmoy-google) [reply kythe/go/extractors/config/smoke/smoke_test.go]: Sorry what caller's `ctx`? In the unit test I'm not sure what the deal is.

Comment (creachadair) [reply kythe/go/extractors/config/smoke/smoke_test.go]: Sorry, I didn't expand the clicky, and misread the code.

Comment (danielmoy-google) [reply kythe/go/extractors/config/smoke/smoke_test.go]: Ok to resolve this using `s/TODO/Background/`

File: kythe/go/extractors/config/smoke/smoke.go [hunk 2]
```diff
@@ -187,7 +187,7 @@ func (g harness) filenamesFromExtraction(repoURI string) (map[string]bool, error
 	ret := map[string]bool{}
 	err = filepath.Walk(tmpOutDir, func(path string, info os.FileInfo, err error) error {
 		if err == nil && filepath.Ext(path) == ".kindex" {
-			cu, err := kindex.Open(context.Background(), path)
```
Comment (danielmoy-google) [inline kythe/go/extractors/config/smoke/smoke.go]: (as a side effect we fix this problematic context.Background() I stuffed in there before too)

**Pull request merged**